x-pack/plugins/monitoring/public/components/logstash/pipeline_viewer/views/config_viewer/statement.js

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { EuiFlexGroup } from '@elastic/eui';
+import { PluginStatement as PluginStatementModel } from '../../models/pipeline/plugin_statement';
+import { CollapsibleStatement } from './collapsible_statement';
+import { IfElement } from '../../models/list/if_element';
+import { PluginStatement } from './plugin_statement';
+
+export class Statement extends React.PureComponent
```

Comment:
I'm not seeing state in this component. Any reason this couldn't be a functional component instead?

Reply:
Same question applies to a few other components in this PR as well.

Reply:
I attempted to convert all stateless classes to functional components in  https://github.com/elastic/kibana/pull/18597/commits/c7342a4cdb14b21a95333e6335e584caba18ed69